File: 800/144A/144A.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n;

    cin >> n;

    vector<int> arr(n);

    for(int i = 0; i < n; i++){
        cin >> arr[i];
    }

    int maxH = *max_element(arr.begin(), arr.end());
    int minH = *min_element(arr.begin(), arr.end());
    int maxIndex = -1;
    int minIndex = -1;

    for(int i = 0; i < n; i++){
        if(arr[i] == maxH){
            maxIndex = i;

            break;
        }
    }

    for(int i = n - 1; i >= 0; i--){
        if(arr[i] == minH){
            minIndex = i;

            break;
        }
    }

    int swaps = maxIndex + (n - minIndex - 1);

    if(maxIndex > minIndex){
        swaps--;
    }

    cout << swaps << '\n';
}

File: 800/144A/144A.py
n = int(input())
arr = list(map(int, input().split()))

maxH = max(arr)
minH = min(arr)

maxIndex = arr.index(maxH)
minIndex = len(arr) - 1 - arr[::-1].index(minH)

swaps = maxIndex + (n - 1 - minIndex)

if maxIndex > minIndex:
    swaps -= 1

print(swaps)